packages/tasit-identity-contract/src/GnosisSafe.js

```diff
@@ -61,59 +61,55 @@ export default class GnosisSafe extends Contract {
     this.#signers = signers;
   };
 
-  transferERC20 = async (tokenAddress, toAddress, value) => {
+  transferERC20 = (tokenAddress, toAddress, value) => {
     const data = this.#utils.encodeFunctionCall(erc20ABI, "transfer", [
       toAddress,
       value,
     ]);
     const etherValue = "0";
-    const action = await this.#executeTransaction(
-      data,
-      tokenAddress,
-      etherValue
-    );
+    const action = this.#executeTransaction(data, tokenAddress, etherValue);
     return action;
   };
 
-  transferNFT = async (tokenAddress, toAddress, tokenId) => {
+  transferNFT = (tokenAddress, toAddress, tokenId) => {
     const fromAddress = this.getAddress();
     const data = this.#utils.encodeFunctionCall(erc721ABI, "safeTransferFrom", [
       fromAddress,
       toAddress,
       tokenId,
     ]);
     const etherValue = "0";
-    const action = await this.#executeTransaction(
-      data,
-      tokenAddress,
-      etherValue
-    );
+    const action = this.#executeTransaction(data, tokenAddress, etherValue);
     return action;
   };
 
-  transferEther = async (toAddress, value) => {
+  transferEther = (toAddress, value) => {
     const data = "0x";
     const etherValue = value;
-    const action = await this.#executeTransaction(data, toAddress, etherValue);
+    const action = this.#executeTransaction(data, toAddress, etherValue);
     return action;
   };
 
-  addSignerWithThreshold = async (newSignerAddress, newThreshold) => {
+  addSignerWithThreshold = (newSignerAddress, newThreshold) => {
     const data = this.#utils.encodeFunctionCall(
       this.getABI(),
       "addOwnerWithThreshold",
       [newSignerAddress, newThreshold]
     );
     const to = this.getAddress();
     const etherValue = "0";
-    const action = await this.#executeTransaction(data, to, etherValue);
+    const action = this.#executeTransaction(data, to, etherValue);
+    return action;
+  };
+
+  #executeTransaction = (data, toAddress, etherValue) => {
+    const tx = this.#prepareTransaction(data, toAddress, etherValue);
+    const provider = this._getProvider();
+    const action = new Action(tx, provider);
```

Comment:
Will an action always need a provider? Maybe not if it's a meta-tx.

Reply:
For now - since all tx are broadcasted to the blockchain - yes. The Action class should be refactored when we start broadcasting to off chain relayers.

Reply:
Okay, that's what I thought. I'll resolve this and we can just keep that in mind for when it comes up again. Thanks.